src/aspire/volume/__init__.py

```diff
@@ -1,16 +1,164 @@
+            res = Volume(self.data * other)
+
+        return res
+
+    # def __truediv__(self, other):
+    #     if isinstance(other, Volume):
+    #         res = Volume(self.data / other.data)
+    #     else:
+    #         res = Volume(self.data / other)
+
+    #     return res
+
+    # # TODO: Open issue for fixing rotated_grids wholesale
+    def _rotated_grids(self, rot_matrices):
+        raise NotImplementedError('Method not currently implemented')
+
+    def project(self, vol_idx, rot_matrices):
+        data = self[vol_idx].T  #RCOPT
+
+        n = rot_matrices.shape[0]
+
+        pts_rot = rotated_grids(self.resolution, rot_matrices)
+
+        # TODO: RCOPT come back and convert these methods internally to C order
+        ## TODO: rotated_grids might as well give us correctly shaped array in the first place,
+        pts_rot = m_reshape(pts_rot, (3, self.resolution**2*n))
+
+        im_f = (1./self.resolution *
+                Plan(self.volume_shape, pts_rot).transform(data))
+
+        im_f = im_f.reshape(-1, self.resolution, self.resolution)
+
+        if self.resolution % 2 == 0:
+            im_f[:, 0, :] = 0
+            im_f[:, :, 0] = 0
+
+        im_f = centered_ifft2(im_f)
+
+        # Looks like we still need to transpose the image axis here for repro
+        im_f = np.swapaxes(im_f, -2, -1)         # RCOPT
+
+        return Image(np.real(im_f))
+
+    def to_vec(self):
+        """ Returns an N x resolution ** 3 array."""
+        return m_reshape(self.data, (self.N,) + (self.resolution**3,))
+        #XXX reshape/flatten?
+
+    @staticmethod
+    def from_vec(vec):
+        """ Returns a Volume instance from a N x resolution**3 array."""
+        N = vec.shape[0]
+        resolution = round(vec.shape[1] ** (1/3))
+        assert resolution**3 == vec.shape[1]
+        data = m_reshape(vec, (N,) + (resolution,)*3)
+        return Volume(data)
+
+    @staticmethod
+    def from_backprojection(im, rot_matrices):
+        """
+        Backproject images along rotation
+        :param im: An Image (stack) to backproject.
+        :param rot_matrices: An n-by-3-by-3 array of rotation matrices \
+        corresponding to viewing directions.
+
+        :return: Volume instance corresonding to the backprojected images.
+        """
+
+        L = im.res
+
+        ensure(im.n_images == rot_matrices.shape[0],
+               "Number of rotation matrices must match the number of images")
+
+        pts_rot = rotated_grids(L, rot_matrices)
+        pts_rot = m_reshape(pts_rot, (3, -1))
+
+        im_f = centered_fft2(im.data) / (L**2)
+        if L % 2 == 0:
+            im_f[:, 0, :] = 0
+            im_f[:, :, 0] = 0
+
+        # yikes
+        im_f = np.swapaxes(im_f, -2, -1)
+        im_f = im_f.flatten()
+
+        plan = Plan(
+            sz=(L, L, L),
+            fourier_pts=pts_rot
+        )
+        vol = np.real(plan.adjoint(im_f)) / L
+
+        return Volume(vol)
+
+    def downsample(self, szout, mask=None):
```

Comment:
Since a `Volume` is always cubic, it may be simpler to just take an `Lout` argument here and expand it to a tuple when calling the `downsample` helper.

Reply:
heh, I had the same idea during one of the other change requests. done. So much cleaner that way...